Examples/Examples/All Examples/SceneKitExample.swift

```diff
@@ -0,0 +1,228 @@
+        transformSimd[2, 3] = m[11].doubleValue
+        transformSimd[3, 0] = m[12].doubleValue
+        transformSimd[3, 1] = m[13].doubleValue
+        transformSimd[3, 2] = m[14].doubleValue
+        transformSimd[3, 3] = m[15].doubleValue
+
+        // Model is using metric unit system: scale x and y from meters to mercator and keep z is in meters.
+        let meterInMercatorCoordinateUnits = try! 1.0 / (Projection.getMetersPerPixelAtLatitude(forLatitude: modelOrigin.latitude, zoom: parameters.zoom))
+        let modelScale = makeScaleMatrix(xScale: meterInMercatorCoordinateUnits, yScale: -meterInMercatorCoordinateUnits, zScale: 1)
+
+        // Translate scaled model to model origin (in web mercator coordinates) and elevate to model origin's altitude (in meters).
+        let origin = try! Projection.project(for: modelOrigin, zoomScale: pow(2, parameters.zoom))
+        var elevation = 0.0
+        if let elevationData = parameters.elevationData, let elevationValue = elevationData.getElevationFor(self.modelOrigin) {
+            elevation = elevationValue.doubleValue
+        }
+        let translateModel = makeTranslationMatrix(tx: origin.x, ty: origin.y, tz: elevation)
+
+        let transform = transformSimd * translateModel * modelScale
+
+        let scnMat = SCNMatrix4(
+            m11: Float(transform[0, 0]),
+            m12: Float(transform[0, 1]),
+            m13: Float(transform[0, 2]),
+            m14: Float(transform[0, 3]),
+            m21: Float(transform[1, 0]),
+            m22: Float(transform[1, 1]),
+            m23: Float(transform[1, 2]),
+            m24: Float(transform[1, 3]),
+            m31: Float(transform[2, 0]),
+            m32: Float(transform[2, 1]),
+            m33: Float(transform[2, 2]),
+            m34: Float(transform[2, 3]),
+            m41: Float(transform[3, 0]),
+            m42: Float(transform[3, 1]),
+            m43: Float(transform[3, 2]),
+            m44: Float(transform[3, 3])
+        )
+        cameraNode.camera!.projectionTransform = scnMat
+
+        // flush automatic SceneKit transaction as SceneKit animation is not running and
+        // there's need to use transform matrix in this frame (not to have it used with delay).
+        SCNTransaction.flush()
+
+        if self.useCPUOcclusion {
+            mtlRenderPassDescriptor.depthAttachment = nil
+            mtlRenderPassDescriptor.stencilAttachment = nil
+            // Example uses depth buffer to occlude model when e.g. behind the hill.
+            // If depth buffer (SCNRenderer.usesReverseZ = false) is not available, or if wished to
+            // to indicate that model is occluded or e.g. implement fade out / fade in model occlusion,
+            // the example here needs to provide CPU side occlusion implementation, too.
+            // TODO: this is blocked on https://github.com/mapbox/mapbox-maps-ios/issues/155
+        }
+        if let colorTexture = mtlRenderPassDescriptor.colorAttachments[0].texture {
```

Comment:
Can we add this as a guard at the top of this function? So 

```swift
guard let colorTexture = mtlRenderPassDescriptor.colorAttachments[0].texture else {
    return
}
```

What would it mean if the model can be rendered without the map?

Reply:
This code won't get called if there is no mtlRenderPassDescriptor.colorAttachments[0].texture.